blocks/bricks/__init__.py

```diff
@@ -146,12 +147,15 @@ def __init__(self, input_dim, output_dim, **kwargs):
         self.output_dim = output_dim
 
     def _allocate(self):
-        self.params.append(shared_floatx_zeros((self.input_dim,
-                                                self.output_dim),
-                           name="W"))
+        W = shared_floatx_zeros((self.input_dim, self.output_dim), name='W')
+        VariableRole.add_role(W, VariableRole.WEIGHTS)
+        self.params.append(W)
+        self.add_auxiliary_variable(W.mean(), name='W_mean')
```

Comment:
I am kind of sceptical that mean weights would be of interest for anybody, weight norms would make more sense.